lib/Brass/API.pm

```diff
@@ -119,18 +119,25 @@ get 'api/pwd/' => sub {
         -cipher => 'Blowfish'
     );
 
-    my $pass;
-    if ($username)
-    {
+    my $pass = query_parameters->get('pass');
+    if ($username) {
+      # update password if new one provided
+      if ($pass) {
+        my $pw = $cipher->encrypt($pass);
+        $username->pwencrypt($pw);
+        $username->update();
+      }
+      else {
         $pass = $cipher->decrypt($username->pwencrypt);
+      }
     }
     else {
-        my $pw = $cipher->encrypt(randompw);
-        my $s = $schema->resultset('Server')->find_or_create({ name => $server });
-        my $u = $schema->resultset('Pw')->create({ server_id => $s->id, username => $param, pwencrypt => $pw, type => $action });
-        $pass = $cipher->decrypt($u->pwencrypt);
+      $pass //= randompw;
```

Comment:
Just a final query, do you think this should be ```||``` instead of ```//``` to prevent accidentally having a "blank" password? Or should that be a feature?

Reply:
hmm.... validation could be worth doing here - something simple like requiring min length : https://metacpan.org/pod/Data::Password::Check looks worth using, we can probably then also use it elsewhere too - now that I think of it - I can also use that in the shared auth password reset code to ensure strong passwords